File: src/main/java/one/digitalinnovation/personapi/person/PersonDTO.java
package one.digitalinnovation.personapi.person;

import io.swagger.annotations.ApiModelProperty;
import one.digitalinnovation.personapi.phone.PhoneDTO;
import org.hibernate.validator.constraints.br.CPF;

import javax.validation.Valid;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Size;
import java.util.List;

public class PersonDTO {
    @ApiModelProperty (notes = "Person ID", example = "1", required = true)
    private Long id;

    @ApiModelProperty(notes = "Person name", example = "joao", required = true)
    @NotEmpty
    @Size(min = 2, max = 100)
    private String firstName;

    @NotEmpty
    @Size(min = 2, max = 100)
    private String lastName;

    @NotEmpty
    @CPF
    private String cpf;

    @NotEmpty
    private String birthDate;

    @Valid
    @NotEmpty
    private List<PhoneDTO> phones;

    public PersonDTO(String firstName, String lastName, String cpf, String birthDate, List<PhoneDTO> phones) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.cpf = cpf;
        this.birthDate = birthDate;
        this.phones = phones;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getCpf() {
        return cpf;
    }

    public void setCpf(String cpf) {
        this.cpf = cpf;
    }

    public String getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(String birthDate) {
        this.birthDate = birthDate;
    }

    public List<PhoneDTO> getPhones() {
        return phones;
    }

    public void setPhones(List<PhoneDTO> phones) {
        this.phones = phones;
    }
}

File: src/main/kotlin/com/elbertribeiro/peopleapi/person/PersonDTO.kt
package com.elbertribeiro.peopleapi.person

import com.elbertribeiro.peopleapi.phone.PhoneDTO
import io.swagger.annotations.ApiModelProperty
import org.hibernate.validator.constraints.br.CPF
import javax.validation.Valid
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.Size

class PersonDTO(
    @field:ApiModelProperty(
        notes = "Person name",
        example = "joao",
        required = true
    )
    var firstName: @Size(min = 2, max = 100) @NotEmpty String?,
    var lastName: @Size(min = 2, max = 100) @NotEmpty String?,
    var cpf: @CPF @NotEmpty String?,
    var birthDate: @NotEmpty String?,
    phones: List<PhoneDTO>
) {
    @ApiModelProperty(notes = "Person ID", example = "1", required = true)
    var id: Long? = null

    private var phones: @Valid @NotEmpty List<PhoneDTO>

    init {
        this.phones = phones
    }

    fun getPhones(): List<PhoneDTO> {
        return phones
    }

    fun setPhones(phones: List<PhoneDTO>) {
        this.phones = phones
    }
}